bin/dlang/install

```diff
@@ -9,22 +9,41 @@ set -eu
 
 beaver=$(dirname $0)/../beaver
 
+remove_old_dmd=0
+
 # Package name deduced based on supplied DMD version
 case "$DMD" in
     dmd*   ) PKG= ;;
     1.*    ) PKG="dmd1=$DMD-$DIST" ;;
     2.*.s* ) PKG="dmd-transitional=$DMD-$DIST" ;;
-    2.*    ) PKG="dmd-bin=$DMD libphobos2-dev=$DMD" ;;
+    2.*    ) if [ $(echo $DMD | cut -d. -f2) -ge 077 ]
+             then
+                PKG="dmd-compiler=$DMD dmd-tools=$DMD libphobos2-dev=$DMD"
+                remove_old_dmd=1
+             else
+                PKG="dmd-bin=$DMD libphobos2-dev=$DMD"
+             fi ;;
     *      ) echo "Unknown \$DMD ($DMD)" >&2; exit 1 ;;
 esac
 
 # Generate the Dockerfile including the DMD_PKG argument and install the
-# relevant DMD
-"$beaver" docker gen-dockerfile \
-        -i beaver.Dockerfile \
-        -I 'ARG DMD_PKG' -I 'ENV DMD_PKG=$DMD_PKG' \
-        -I 'RUN apt update && apt -y install --allow-downgrades $DMD_PKG' \
-        -o beaver.Dockerfile.generated
+# relevant DMD. `-I` arguments don't seem to work with shell expansion nicely,
+# thus using plain if block and copying of the command.
```

Comment:
What is not working exactly? Also, you can use `apt -y install pgk to-remove-` (the trailing `-` will remove the package, even if the command is `install`.

Reply:
> What is not working exactly?

I tried using `OPTIONAL_I` variable which would expand to either `-I 'something'` or empty string but it resulted in literally trying to put `'` symbols into generated dockerfile. No idea why.

> Also, you can use apt -y install pgk to-remove- (the trailing - will remove the package, even if the command is install.

TIL. That will make things much simpler indeed.